katportalclient/client.py

```diff
@@ -619,6 +620,236 @@ def schedule_blocks_assigned(self):
+            pointing details.
+        timestamp: float
+            The unix timestamp (UTC) of the time of the reference observer
+            used to calculate the pointing details.
+        """
+        self._reference_observer_config['longitude'] = longitude
+        self._reference_observer_config['latitude'] = latitude
+        self._reference_observer_config['altitude'] = altitude
+        self._reference_observer_config['timestamp'] = timestamp
+
+    @tornado.gen.coroutine
+    def config_label_for_subarray(self, sub_nr):
+        """Get the config label for the specified sub_nr. Returns None if
+        the subarray has an empty config_label. An active subarray will
+        always have a config_label.
+
+        Parameters
+        ----------
+        sub_nr: int
+            The subarray's config label to get, can be 1,2,3,4
+
+        Returns
+        -------
+        str:
+            A csv string that is used as a config_label in the CAM system.
+            The config_label is used to select the version of the catalogue
+            files when calculating the pointing details.
+        """
+        url = self.sitemap['subarray_sensor_values'] + '/config_label'
+        response = yield self._http_client.fetch(url)
+        config_label_sensor_result = json.loads(response.body)[0]
+        raise tornado.gen.Return(config_label_sensor_result.get('value'))
+
+    @tornado.gen.coroutine
+    def _get_target_descriptions(
+            self, targets, longitude, latitude, altitude,
+            timestamp, config_label):
+        """
+        Return a list of targets with their pointing information calculated
+        by using the longitude, latitude, altitude as the reference observer
+        and timestamp as the time of the reference observer.
+        Using the config_label to select a specific version of the catalogues
+        files.
+
+        Parameters
+        ----------
+        targets: str
+            A CSV list of target names to retrieve the pointing details.
+        longitude: float
+            The longitude of the reference observer used in calculating the
+            pointing details.
+        latitude: float
+            The latitude of the reference observer used in calculating the
+            pointing details.
+        altitude: float
+            The altitude of the reference observer used in calculating the
+            pointing details.
+        timestamp: float
+            The unix timestamp (UTC) of the time of the reference observer
+            used to calculate the pointing details.
+        config_label: str
+            The config_label to use to select the version of the catalogue
+            files when calculating the pointing details.
+
+        Returns
+        -------
+        list:
+            A list of dictionaries containing pointing information calculated
+            taking the reference observer into account.
+        """
+        url = self.sitemap['target_descriptions'] + (
```

Comment:
Thinking about this a bit more the problem is not necessarily with GET but more to do with how we use the end point to embed the variables, it is trying to look like REST but it is not. GET params would be ok here.